ibc-apps/ics721-nft-transfer/types/src/msgs/transfer.rs

```diff
@@ -49,19 +47,18 @@ pub struct MsgTransfer {
 }
 
 impl TryFrom<RawMsgTransfer> for MsgTransfer {
-    type Error = NftTransferError;
+    type Error = DecodingError;
 
     fn try_from(raw_msg: RawMsgTransfer) -> Result<Self, Self::Error> {
-        let timeout_height_on_b: TimeoutHeight = raw_msg
-            .timeout_height
-            .try_into()
-            .map_err(ContextError::from)?;
+        let timeout_height_on_b: TimeoutHeight = raw_msg.timeout_height.try_into()?;
 
         let timeout_timestamp_on_b: TimeoutTimestamp = raw_msg.timeout_timestamp.into();
 
         // Packet timeout height and packet timeout timestamp cannot both be unset.
         if !timeout_height_on_b.is_set() && !timeout_timestamp_on_b.is_set() {
-            return Err(ContextError::from(PacketError::MissingTimeout))?;
+            return Err(DecodingError::missing_raw_data(
+                "neither timeout height nor timeout timestamp set",
```

Comment:
I would try to stick with the convention of how the different error messages are worded per each error classification. 
```suggestion
                "missing timeout height or timeout timestamp",
```

Reply:
Makes sense. Thanks for highlighting that.
[e24802d](https://github.com/cosmos/ibc-rs/pull/1335/commits/e24802d435ced8cf18cae03c7999dae9e91d4285)